src/reporter.cc

```diff
@@ -71,46 +78,109 @@ void MetadataReporter::ReportMetadata() {
 namespace {
 
 void SendMetadataRequest(std::vector<json::value>&& entries,
-                         const std::string& endpoint,
+                         const std::string& batch_uri,
+                         const std::string& publish_path,
                          const std::string& auth_header,
                          const std::string& user_agent,
                          bool verbose_logging)
     throw (boost::system::system_error) {
-  json::value update_metadata_request = json::object({
-    {"entries", json::array(std::move(entries))},
-  });
 
-  if (verbose_logging) {
-    LOG(INFO) << "About to send request: POST " << endpoint
-              << " User-Agent: " << user_agent
-              << " " << *update_metadata_request;
+  if (entries.size() == 1) {
+    // Add a copy of the single entry, except for the `views` field. This allows
+    // us to sent a request to the batch endpoint when we have a single request.
```

Comment:
Is there any issue that this blob is sent LAST, would that not overwrite the previous entry?

Reply:
This will hit the API as separate requests, so both  requests will make it through.
Given that views is missing completely, that should not be overwritten.